lib/validation.js

```diff
@@ -81,25 +83,25 @@ function compileSchemasForValidation (context, compile, isCustom) {
     }
     context[headersSchema] = compile({ schema: headersSchemaLowerCase, method, url, httpPart: 'headers' })
   } else if (Object.hasOwnProperty.call(schema, 'headers')) {
-    throw new Error('headers schema is undefined')
+    warning.emit('FSTDEP015', 'headers')
```

Comment:
the user does not know the route that triggers it - it would hard to debug without it

(the error was carrying the information)

Reply:
commit with the proposed message: 5f2465a04ead6e8d174837bc3094a13ba762bac7